x/dsmr/node.go

```diff
@@ -263,6 +293,12 @@ func (n *Node[T]) Accept(ctx context.Context, block Block) error {
 			}
 		}
 	}
+	// update the validity window with the accepted block.
+	n.validityWindow.Accept(block)
+
+	return n.storage.SetMin(block.Tmstmp, chunkIDs)
+}
 
-	return n.storage.SetMin(block.Timestamp, chunkIDs)
+func (n *Node[T]) GetExecutionBlock(ctx context.Context, blkID ids.ID) (validitywindow.ExecutionBlock[*ChunkCertificate], error) {
```

Comment:
we can defer this to the VM to maintain a chain index for us that we pass in - similar to how we handle this in the existing `chain/` package

Reply:
np